src/librustdoc/html/render/print_item.rs

```diff
@@ -954,13 +954,13 @@ fn item_trait(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, t: &clean:
     if !t.is_object_safe(cx.tcx()) {
         write_section_heading(
             w,
-            "Object Safety",
+            "Trait-Object Safety",
             "object-safety",
             None,
             &format!(
-                "<div class=\"object-safety-info\">This trait is <b>not</b> \
+                "<div class=\"object-safety-info\">This trait is <strong>not</strong> compatible with \
                 <a href=\"{base}/reference/items/traits.html#object-safety\">\
```

Comment:
You should leave a `FIXME(dyn_compat_rename): Update the URL fragment once the changes in <https://github.com/rust-lang/reference/pull/1512> reach stable`.